include/mbgl/storage/default_file_source.hpp

```diff
@@ -127,6 +143,8 @@ class DefaultFileSource : public FileSource {
     const std::unique_ptr<util::Thread<Impl>> thread;
     const std::unique_ptr<FileSource> assetFileSource;
     const std::unique_ptr<FileSource> localFileSource;
+    std::string cachedBaseURL;
```

Comment:
You probably need to initialize this one with `mbgl::util::API_BASE_URL`, otherwise if you call the getter before setting, the result will be inconsistent.